docs/api-architecture.md

```
@@ -0,0 +1,1051 @@
+
+---
+
+## Table of Contents
+
+1. [Global Architecture](#global-architecture)
+2. [Core Routes (Legacy `/api/`)](#core-routes-legacy-api)
+   - [/api/organizations](#apiorganizations)
+   - [/api/projects](#apiprojects)
+   - [/api/stats](#apistats)
+   - [/api/tech-stack](#apitech-stack)
+   - [/api/tech-stack/:slug](#apitech-stackslug)
```

markdownlint-cli2 (0.18.1)

Link fragments should be valid

(MD051, link-fragments)